esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -0,0 +1,188 @@
+        # colorbar size often does not fit nicely
+        # https://stackoverflow.com/questions/18195758/set-matplotlib-colorbar-size-to-match-graph
+        # cbar_kwargs={'fraction': .021}
+    )
+
+    ax.coastlines()
+    lons = dataarray.lon.values
+    lats = dataarray.lat.values
+    longitude_formatter = LongitudeFormatter()
+    latitude_formatter = LatitudeFormatter()
+    if 'xticks' in cfg.keys():
+        ax.set_xticks(cfg['xticks'])
+    else:
+        ax.set_xticks(np.arange(np.floor(lons.min()), np.ceil(lons.max()), 10), crs=proj)
+    if 'yticks' in cfg.keys():
+        ax.set_yticks(cfg['yticks'])
+    else:
+        ax.set_yticks(np.arange(np.floor(lats.min()), np.ceil(lats.max()), 10), crs=proj)
+    ax.xaxis.set_ticks_position('both')
+    ax.yaxis.set_ticks_position('both')
+    ax.xaxis.set_major_formatter(longitude_formatter)
+    ax.yaxis.set_major_formatter(latitude_formatter)
+    ax.set_xlabel('')
+    ax.set_ylabel('')
+
+    title = title_pattern.format(metric=metric, period=period)
+    ax.set_title(title)
+
+    filename_plot = get_plot_filename(
+        filename_part, cfg)
+    fig.savefig(filename_plot, dpi=300, bbox_inches='tight')
+    plt.close(fig)
+
+    filename_data = get_diagnostic_filename(
+        filename_part, cfg, extension='nc')
+    dataarray.to_netcdf(filename_data)
+
+    log_provenance(title, filename_plot, cfg, ancestors)
+    log_provenance(title, filename_data, cfg, ancestors)
+
+
+def visualize_and_save_temperature(temperature: 'xr.DataArray',
+                                   cfg: dict,
+                                   ancestors: list):
+    """Wrapper for mapplot: absolute temperature"""
+    title_pattern = 'Weighted {metric} temperature \n{period} ($\degree$C)'
+    filename_part = 'temperature_change_weighted_map'
+    mapplot(temperature, cfg, title_pattern, filename_part, ancestors, cmap='Reds')
+
+
+def visualize_and_save_temperature_difference(temperature_difference: 'xr.DataArray',
+                                              cfg: dict,
+                                              ancestors: list):
+    """Wrapper for mapplot: temperature change"""
+    title_pattern = 'Difference: weighted minus unweighted {metric} temperature\n{period} ($\degree$C)'
+    filename_part = 'temperature_change_difference_map'
+    mapplot(temperature_difference, cfg, title_pattern, filename_part, ancestors, center=0)
+
+
+def calculate_percentiles(data: 'xr.DataArray',
+                          percentile: float,
+                          weights: dict = None):
+    """Calculate (weighted) percentiles.
+
+    Calculate the (weighted) percentiles for the given data.
+
+    Percentiles is a list of values between 0 and 100.
+
+    Weights is a dictionary where the keys represent the model members,
+    and the values the weights.
+    If `weights` is not specified, the non-weighted percentiles are calculated.
+
+    Returns a DataArray with 'percentiles' as the dimension.
```

Comment:
This return description is outdated, I think. 

This function is almost the same as the one in the timeseries plot. The only difference seems to be whether there is one or multiple percentiles. Would it make sense to move the multiple percentiles one to the utils directory as well? I think that one should work as well if you use a lenth-1 array for the percentiles, no?